### ISSUE TITLE
GO:0003961 O-acetylhomoserine aminocarboxypropyltransferase  and O-acetylhomoserine sulfhydrylase activity (GO:0051009)

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-ontology#28681]
SGD and PomBase had confusion over the term for homocysteine synthase. We have both used 
O-acetylhomoserine aminocarboxypropyltransferase activity (GO:0003961)
instead of
O-acetylhomoserine sulfhydrylase activity (GO:0051009)
We need to add "homocysteine synthase" as a synonym of O-acetylhomoserine sulfhydrylase activity (GO:0051009)

@robnash  also spotted 
"Additionally, I think that GO:0003961 O-acetylhomoserine aminocarboxypropyltransferase activity appears to perhaps be a parent of GO:0051009 O-acetylhomoserine sulfhydrylase activity, as it lists OAH sulfhydrylase activity as a synonym, and although the reactants and products listed are different it says in the definition second sentence "Also reacts with other thiols and H2S, producing homocysteine or thioethers.” (which is what is used in these papers). So there may be some GO clean up here"

@raymond91105 would you be happy to look at this one?

**Comment by ValWood** [comment]
In case its helpful, Rob's notes:

 In both the paper used for the MET17 annotation (PMID: 7765825) and the paper used for the HSU1 annotation (PMID:36379252), the inefficient step-sister (homocysteine synthase), the assay uses OAH and H2S and produces homocysteine. 

Here is an excerpt from the HSU1 paper (where MET15 = MET17):
"We then carried out in vitro reactions using the canonical OAH and H2S substrates as well as PLP, a cofactor critical for Met15 activity. Yll058w was able to catalyze homocysteine production significantly above background, albeit much less efficiently than Met15, with the Met15 reactions yielding ∼9.9 times as much homocysteine at the 15 min time point ([Fig. S10](https://www.jbc.org/article/S0021-9258(22)01140-1/fulltext#appsec1)A). We then measured enzyme kinetic parameters by performing analogous reactions with varying amounts of the limiting substrate, OAH ([Fig. 3](https://www.jbc.org/article/S0021-9258(22)01140-1/fulltext#gr3)E). Yll058w activity was consistently above background and increased with increasing OAH concentrations (Vmax: 1.62; Km: 4.29), although remaining less efficient than Met15 activity at all OAH concentrations (Vmax: 89.21; Km: 8.03). The putative catalytic dead mutant Yll058w (K376A) had no detectable activity. We therefore conclude that YLL058W encodes an inefficient homocysteine synthase capable of catalyzing the same reaction as Met15.”

and 

"It then adds the homocysteine-generating reaction that we observe in our in vitro assay, which uses H2S and O-acetylhomoserine as substrates and yields homocysteine, acetate, and hydrogen as products"
